src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,73 @@
+package com.gmail.woosay333.onlinebookstore.exception.handler;
+
+import com.gmail.woosay333.onlinebookstore.exception.BookIsbnAlreadyExistsException;
+import com.gmail.woosay333.onlinebookstore.exception.DataProcessingException;
+import com.gmail.woosay333.onlinebookstore.exception.EntityNotFoundException;
+import com.gmail.woosay333.onlinebookstore.exception.RegistrationException;
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```

Comment:
Why logic of error response here is different from other methods, is it possible to make it same?

Reply:
I hope it's a little better this way :)